tests/t5/test_modeling_tf_t5.py

```diff
@@ -524,6 +524,35 @@ def test_greedy_generate(self):
 
         self.assertListEqual(expected_output_string, output_strings)
 
+    @slow
+    def test_sample_xla_generate_simple(self):
+        model = TFT5ForConditionalGeneration.from_pretrained("t5-small")
+        tokenizer = T5Tokenizer.from_pretrained("t5-small")
+
+        sentence = "Translate English to German: Today is a beautiful day."
+        input_ids = tokenizer(sentence, return_tensors="tf", padding=True).input_ids
+        # XLA reorder ops, which causes operations like FP matmul to have slightly different results, causing
+        # divergences in generate -- especially with sampling.
+        expected_output_string = ["Heute ist ein schöner Tag."]
+        expected_output_string_xla = ["Heute ist ein schöne Tage."]
```

Comment:
Small comment but "schöner" is correct here - I hope it's just sampling issues and not a sign of a bug that the XLA version gets it wrong!

Reply:
Could you try, as a kind of stupid manual test, asking it to translate a batch of sentences from English to Portuguese and make sure that even if they're different, the quality of the XLA ones is similar to the quality of the manual ones? Numerical bugs can be very annoying to catch, but if the quality is similar then that would make me confident that the XLA implementation is not worse.